### DISCUSSION TITLE
Button Mapping

### DISCUSSION THREAD

**Discussion by DanAlexG** [tsunamods-codes/Junction-VIII#7] [General]
- Controls were being a challenge so ended up cleaning the game and J8 
- Went for clean install of game + J8 1.0.0.29 (Ensured no remains on the filesystem before install)
- Trying to get custom mapping, ensured defaults are set on game
- Set controls to my preference in J8, saved as a new profile
- Mapping set does not reflect the same way ingame, unsure if it made changes and just maps differently from descriptions or simply does not respect the mappings.

**Comment by julianxhokaxhiu** [comment]
Sorry can you please explain which point are you trying to make here?

**Comment by DanAlexG** [comment]
- My apologies for not being clear enough, basically whatever custom mapping of controls I set in J8 are not respected within FF8 once launched. It appears to have issues with it.

Edit for clarity: 
FF8 Steam version, not using steam input. 
I set the custom controls in J8, launch the game, the controls do not match what I set in J8 after loading the save.

**Reply by julianxhokaxhiu** [reply]
Thanks I will have a look at this.

**Reply by DanAlexG** [reply]
Appreciate it, I have posted on discord the exact output I get with the settings screenshot.
![settings_output](https://github.com/user-attachments/assets/ccb1b8e1-1617-4b6b-a7e3-2d845f6abe89)
From log it seems to be copying the correct file:
13:43:19|INFO|AppUI.Classes.GameLauncher|    Copying C:\Users\danie\AppData\Local\Programs\Junction VIII\Resources\Controls\dan.cfg -> C:\Users\danie\Documents\Square Enix\FINAL FANTASY VIII Steam\ff8input.cfg ..

**Reply by DanAlexG** [reply]
The input file that generates from the settings above:

Keyboard
1. "Menu"   32
2. "Select"     45
3. "Misc"     30
4. "Exit"     17
5. "Toggle"   16
6. "Trigger"  18
7. "RotLt"    44
8. "RotRt"    46
9. "Start"    31
10. "Select"   33
11. "Up"       200
12. "Down"     208
13. "Left"     203
14. "Right"    205
Joystick
1. "Menu"   227
2. "Select"     225
3. "Misc"     224
4. "Exit"     226
5. "Toggle"   228
6. "Trigger"  229
7. "RotLt"    230
8. "RotRt"    231
9. "Start"    232
10. "Select"   233
11. "Up"       252
12. "Down"     253
13. "Left"     254
14. "Right"    255

**Reply by DanAlexG** [reply]
Thanks for the help! Can confirm in latest canary both mapping and default control profile persistence are fixed :) Cheers!

**Reply by julianxhokaxhiu** [reply]
Glad to hear that. Thanks a lot for the feedback provided! Enjoy :)
